hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRequestFutureImpl.java

```diff
@@ -842,6 +849,18 @@ private void resubmit(ServerName oldServer, List<Action> toReplay, int numAttemp
         backOffTime, true, null, -1, -1));
     }
 
+    long remainingTime = getRemainingTime();
+    // 1 is a special value meaning exceeded and 0 means no timeout
+    if (remainingTime > 1 && backOffTime > remainingTime) {
```

Comment:
I added this handling here as well. We have similar logic in RpcRetryingCallerImpl. I think it's important to add when adding throttling wait interval, because depending on the level of quota saturation the wait time could easily be higher than operation timeout.